### PR TITLE
[core][ci] Update state api scale test

### LINKED ISSUE #35460: [release-tests] stress_test_state_api_scale.aws failed
### What happened + What you expected to happen

stress_test_state_api_scale.aws has been failing and is blocking 2.5 release. Please help review and resolve asap. 

I kicked off a bisect job here: https://buildkite.com/ray-project/release-tests-bisect/builds/158#_

### Versions / Dependencies

master

### Reproduction script

https://buildkite.com/ray-project/release-tests-branch/builds/1658#01882a6d-ba78-4de5-bc7d-ace4c9863af8

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
We are using a signal block to block the execution of tasks because tasks states were not persistent when they finished before (so we need to have X number of tasks running to simulate a large task response). 

However, this is no longer true.

## Related issue number

Closes https://github.com/ray-project/ray/issues/35460

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/stress_tests/stress_tests_state_api_compute.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
+region: us-west-2
+
+max_workers: 2
```
Comment (rickyyx) [inline release/nightly_tests/stress_tests/stress_tests_state_api_compute.yaml:4]: was 6

**Comment by rickyyx** [comment]
Recent run: https://buildkite.com/ray-project/release-tests-pr/builds/40860#01887967-557d-4960-a7e5-d6e2845d9705

**Comment by rickyyx** [comment]
Added a CI test that runs the release test's smoke test.

**Comment by rkooo567** [comment]
```
test_state_api_scale_smoke
```
failure seems related

**Comment by rickyyx** [comment]
Release run: https://buildkite.com/ray-project/release-tests-pr/builds/40904#01887a60-047d-4547-9951-c7be6ef58804

**Comment by rkooo567** [comment]
test is still failing (python medium a-j)

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**